providers/implementations/rands/drbg.c

```diff
@@ -922,3 +922,30 @@ int ossl_drbg_set_ctx_params(PROV_DRBG *drbg, const OSSL_PARAM params[])
         return 0;
     return 1;
 }
+
+/* Confirm digest is allowed to be used with a DRBG */
+int ossl_drbg_verify_digest(const EVP_MD *md)
+{
+#ifdef FIPS_MODULE
+    /* FIPS 140-3 IG D.R limited DRBG digests to a specific set */
+    static const char *const allowed_digests[] = {
```

Comment:
should this be conditional based on the security_checks property?

Reply:
Then we could potentially test this - althought it is painful

Reply:
That's a good question.  I don't know.

Reply:
It's easy to manually test by taking out the FIPS version lines from the evp_test data file and run it.  It fails wonderfully.

Reply:
Having this as conditional seems like a good idea. This is a potential breaking change. Shouldn't we be giving people the option: break your app or comply with FIPS?

Reply:
I've made it conditional but on a new variable.